core/src/main/java/com/linecorp/armeria/server/annotation/processor/DocumentationProcessor.java

```diff
@@ -0,0 +1,186 @@
+        final Properties properties = readProperties(className);
+        final String docComment = processingEnv.getElementUtils().getDocComment(method);
+        if (docComment == null || !docComment.contains("@param")) {
+            return;
+        }
+        final List<List<String>> lines = Arrays.stream(docComment.split("\\R"))
+                                               .filter(line -> !line.trim().isEmpty())
+                                               .map(line -> Arrays.stream(line.split("\\s"))
+                                                                  .filter(word -> !word.trim().isEmpty())
+                                                                  .collect(toImmutableList()))
+                                               .collect(toImmutableList());
+        method.getParameters().forEach(param -> {
+            final StringBuilder stringBuilder = new StringBuilder();
+            JavaDocParserState state = JavaDocParserState.SEARCHING;
+            for (List<String> line : lines) {
+                final List<String> subLine;
+                if ((line.size() < 3 ||
+                     !"@param".equals(line.get(0)) ||
+                     !param.getSimpleName().toString().equals(line.get(1))) &&
+                    state == JavaDocParserState.SEARCHING) {
+                    continue;
+                } else if (state == JavaDocParserState.IN_DESCRIPTION &&
+                           line.size() > 0 &&
+                           "@param".equals(line.get(0))) {
+                    break;
+                } else if (state == JavaDocParserState.SEARCHING) {
+                    subLine = line.subList(2, line.size());
+                    state = JavaDocParserState.IN_DESCRIPTION;
+                } else {
+                    subLine = line;
+                }
+                for (String word : subLine) {
+                    stringBuilder.append(word);
+                    stringBuilder.append(' ');
+                }
+            }
+            setProperty(properties, method, param, stringBuilder.toString().trim());
+        });
+    }
+
+    private void setProperty(Properties properties,
+                             ExecutableElement method,
+                             VariableElement parameter,
+                             String description) {
+        final String methodName = method.getSimpleName().toString();
+        final String parameterName = parameter.getSimpleName().toString();
+        properties.setProperty(methodName + "." + parameterName, description);
+    }
+
+    /**
+     * Creates the file name used in the rest api documentation properties files.
+     * @param className The class name used for generating the file name.
+     * @return The used file name.
+     */
+    public static String getFileName(String className) {
+        return className + "-generated-docs.properties";
```

Comment:
Perhaps we should name the file in a similar manner with the version properties files, i.e. `META-INF/com.linecorp.armeria.docstrings.annotated.properties`

Reply:
We could also refrain from generating multiple `.properties` files but appending to an existing one, so it's easier to look for when scanning them.

Reply:
Sadly, it is not possible to use the same file. The preprocessor has multiple rounds (java and kotlin, I believe) and you can open a file only once for writing, globally...

https://bugs.openjdk.java.net/browse/JDK-6501983 (evaluated as not an error...)

But I will change the file name

Reply:
Also, the `getFileName` method is static and public which allows you to just feed the qualified class name into this method and you have the path for the resource loader in the doc service. I think it is pretty easy to get the file name...